flatmap/expand.go

```diff
@@ -114,6 +114,21 @@ func expandArray(m map[string]string, prefix string) []interface{} {
 }
 
 func expandMap(m map[string]string, prefix string) map[string]interface{} {
+	// Submaps may not have a '%' key, so we can't count on this value being
+	// here. If we don't have a count, just procede as if we have have a map.
+	if count, ok := m[prefix+"%"]; ok {
+		num, err := strconv.Atoi(count)
+		if err != nil {
+			panic(err)
+		}
+
+		// If the number of elements in this map is 0, then return an
+		// empty map as there is nothing to expand.
+		if num == 0 {
+			return map[string]interface{}{}
+		}
+	}
+
```

Comment:
I would do this simpler... As we don't really need `num` here, why not just something like this:

```
if count, ok := m[prefix+"%"]; ok && count == "0" {
  return map[string]interface{}{}
}
```

Other then that LGTM! 😀